### ISSUE TITLE
Precompiled kernel 4.18.20-4 boots to emergency shell on surface pro 4

### ISSUE THREAD

**Issue by SnowMB** [dmhacker/arch-linux-surface#15]
Hey,

I cloned your repository and installed the precompiled kernel from your latest release as instructed in the readme. However when I try to boot from the kernel I get dropped into the emergency shell with the following message:

```
Warning: /lib/modules/4.18.20-4-surface/modules.devname not found - ignoring
Starting version 240
arch: clean, 1428820/9732096 files, 9407575/38912000 blocks
mount: /new_root: unknown filesystem type 'ext4'.
You are now being dropped into an emergency shell.
sh: can't access tty: job control turned off
[rootfs ]#

```

Where did I go wrong?

**Comment by dmhacker** [comment]
I don't think this is an issue with the pre-compiled kernels, since it's working fine for me. Maybe try checking out these threads?

https://bbs.archlinux.org/viewtopic.php?id=215432
https://bbs.archlinux.org/viewtopic.php?id=212129

**Comment by NitroBAY** [comment]
Hello,
I got the same problem both with compiled and pre-compiled kernels.
I've checked the kernel but it didn't help.
I use reefind dual boot Windows 10/arch last version. When I choose from reefind to use the regular kernel things work, it does not with surface's kernel

**Comment by realSpok** [comment]
Same problem here on a surface pro 4. I was using the AUR repo, and tried to update to this kernel, but I end up on the same error as OP.
Thankfull the old kernel is still available in my Refind menu.

**Comment by realSpok** [comment]
Ok, I added this line to my refind.conf, in /boot:

"Boot new kernel"				 "root=PARTUUID=751cc960-c067-014c-b5d2-9bc1f414dc12 add_efi_memmap rootfstype=ext4 initrd=/boot/intel-ucode.img initrd=/boot/initramfs-linux-surface.img loglevel=3 vga=current"


Booting from this solves the issue.

**Comment by NitroBAY** [comment]
Nice people seeing this issue don't forget to # blkid to get your root partition PARTUUID

**Comment by dmhacker** [comment]
To chime in, on my system, I am not using rEFInd, but I do have intel-ucode installed. That being said, I'm not sure how much help I will be with respect to this issue. To all of you that have not installed the intel-ucode binary, I'd recommend you do so, because it contains necessary microcode for the processor. You can do this through the `setup.sh` script in the root of this repository.

**Comment by SnowMB** [comment]
I also use refind for the touch support in the boot menu. Will try the suggested solutions and report back. Thanks everyone for Gavin a look at this.

**Comment by SnowMB** [comment]
Sorry for reporting back so late. 

Modifying `/boot/refind_linux.conf` like @realSpok  suggested fixed this for me:
```
"Boot extra"                  "root=UUID=... rw quiet resume=UUID=... add_efi_memmap rootfstype=ext4 initrd=/boot/intel-ucode.img initrd=/boot/initramfs-linux-surface.img"
```

But interestingly I can't boot the default kernel with this options. There I have to use the default option:
```
"Boot with standard options"  "root=UUID=... rw quiet resume=UUID=..."
```

**Comment by ricardo17coelho** [comment]
![20191117_124431](https://user-images.githubusercontent.com/37732431/69657734-279f6200-107b-11ea-8daa-cbc4de02a936.jpg)

can someone help me please?

**Comment by ricardo17coelho** [comment]
i'm using refind too.
my refind.conf looks like:

![15747898337812200644619238097819](https://user-images.githubusercontent.com/37732431/69658088-d2178500-107b-11ea-92e4-d4bdcc5c1baf.jpg)


![1574789882594860234918407896258](https://user-images.githubusercontent.com/37732431/69658199-0a1ec800-107c-11ea-8351-d4824c9fc494.jpg)
